### PR TITLE
Fix #99: Allow int or stringable object as userId

### LINKED ISSUE #99: Add support `$userId` as stringable object
`Manager` is inconsistent in part or `$userId` method argument:
 
- `userHasPermission` allow `$userId` as string or integer;
- Another methods allow only string.

Suggest allow in all methods of `Manager` `$userId` as string, integer or stringable object.

Why stringable object? 
In application user ID can be value object.

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | #99

### CODE REVIEW

File: src/CompositeRule.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ final class CompositeRule implements RuleInterface
 
     /**
      * @param string $operator Operator to be used. Could be `CompositeRule::AND` or `CompositeRule::OR`.
-     * @param RuleInterface[] $rules Array of rule instances.
+     * @param array|RuleInterface[] $rules Array of rule instances.
```
Comment (vjik) [inline src/CompositeRule.php:39]: `$rules` can only be array of `RuleInterface`. Why added array of mixed values?

File: src/Manager.php [hunk 2]
```diff
@@ -323,18 +324,36 @@ public function getPermissionsByRoleName(string $roleName): array
...
+     * @param mixed $userId
+     *
+     * @return string
+     */
+    private function ensureStringUserId($userId): string
+    {
+        if (!is_string($userId) && !is_int($userId) && !(is_object($userId) && method_exists($userId, '__toString'))) {
+            $type = is_object($userId) ? get_class($userId) : gettype($userId);
+            throw new InvalidArgumentException(
+                sprintf('userId must be a string or an int, %s given.', $type)
```
Comment (vjik) [inline src/Manager.php:351]:
```suggestion
                sprintf('User ID must be a string, an integer or an object with method "__toString()", %s given.', $type)
```

File: src/CompositeRule.php (hunk 1 above)
Comment (samdark) [reply src/CompositeRule.php]: Fixed.

File: src/Manager.php (hunk 2 above)
Comment (samdark) [reply src/Manager.php]: Fixed.

**Pull request merged**